packages/core/src/components/FormLabel/FormLabel.jsx

```diff
@@ -43,6 +54,7 @@ export class FormLabel extends React.PureComponent {
         <span className={labelTextClasses}>{this.props.children}</span>
         {this.errorMessage()}
         {this.hint()}
+        {this.requirementLabel()}
```

Comment:
I'm interpreting the design differently. I'm under the impression that the `requirementLabel` is the first child of the "hint" element, and is followed by any other hint text.

Pseudo code:

```jsx
<span>{this.props.requirementLabel} {this.props.hint}</span>
```